.changeset/slimy-lions-teach.md

```diff
@@ -0,0 +1,75 @@
+---
+"thirdweb": minor
+---
+
+New PayEmbed modes and revamp TransactionButton flow
+
+You can now configure the PayEmbed component to build 3 different flows:
+
+- Fund wallets: Inline component that allows users to buy any currency. (default)
+
+```tsx
+<PayEmbed
+  client={client}
+  payOptions={{
+    mode: "fund_wallet",
+  }}
+/>
+```
+
+- Direct payments: Take payments from Fiat or Crypto directly to your seller wallet.
+
+```tsx
+<PayEmbed
+  client={client}
+  payOptions={{
+    mode: "direct_payment",
+    paymentInfo: {
+      sellerAddress: "0x...",
+      chain: base,
+      amount: "0.1",
+    },
+    metadata: {
+      name: "Black Hoodie (Size L)",
+      image: "https://example.com/image.png",
+    },
+  }}
+/>
+```
+
+- Transaction payments: Let your users pay for onchain transactions with fiat or crypto on any chain.
+
+```tsx
+<PayEmbed
+  client={client}
+  payOptions={{
+    mode: "transaction",
+    transaction: claimTo({
```

Comment:
Can we make this a function similar to `TransactionButton` that returns a transaction object - so that it does not create a new transaction object on every re-render of the component

Reply:
yep good call

Reply:
so i tried this, and we actually can't have it as a function at all because we need the information about this transaction at first render time (value, chain, erc20value etc) - that means i would execute the function right away, which is very confusing as a dev (like i might throw in there because of some state, which would throw immediately)

so i think its actually better to leave it as a straight prepared transaction to FORCE you to conditionally render this once you have everything you need to prepare the transaction